prime-router/docs/design/design/deduplication/deduplication.md

```diff
@@ -0,0 +1,197 @@
+  - Set flag or enter a workflow to check if entire report is duplicate.
+      - If the entire report is found to be duplicate, this will be logged as an error. See [example error](#example-submission-history-api-entire-report-is-duplicate) below.
+
+##### Example: Submission History API, Some Item(s) in Batched Report are Duplicates 
+
+```json
+{
+    "id": "849b3151-25f7-41f3-b19b-fd8ad47bae18",
+    "submissionId": 63,
+    "overallStatus": "Delivered",
+    "timestamp": "2025-01-22T00:42:10.404Z",
+    "sender": "development.dev-elims",
+    "reportItemCount": 2,
+    "errorCount": 0,
+    "warningCount": 1,
+    "httpStatus": 201,
+    "destinations": [
+        {
+            "organization": "FOR DEVELOPMENT PURPOSES ONLY",
+            "organization_id": "development",
+            "service": "DEV_ENRICHMENT_FHIR",
+            "itemCount": 1,
+            "itemCountBeforeQualityFiltering": 0,
+            "sending_at": "2025-01-22T00:43:00.000Z",
+            "filteredReportRows": [],
+            "filteredReportItems": [],
+            "sentReports": [
+                {
+                    "reportId": "dbc9ac68-77a0-4e3e-8d8b-2d665b8fd92c",
+                    "externalName": "none-c7c7e0fe-1951-4031-9d17-2895fc03c974-20250121164303.fhir",
+                    "createdAt": "2025-01-22T00:43:04.252Z",
+                    "itemCount": 1
+                }
+            ],
+            "downloadedReports": []
+        }
+    ],
+    "reportId": "849b3151-25f7-41f3-b19b-fd8ad47bae18",
+    "topic": "full-elr",
+    "errors": [],
+    "warnings": [
+        {
+            "scope": "item",
+            "indices": [
+                1
+            ],
+            "trackingIds": [
+                "849b3151-25f7-41f3-b19b-fd8ad47bae18"
+            ],
+            "message": "Report could not processed. Reason: Duplicate item was detected and removed.",
+            "errorCode": "DUPLICATE_ITEM"
+        }
+    ],
+    "destinationCount": 1,
+}
+```
+
+##### Example: Submission History API, Entire Report is Duplicate
+
+```json
+{
+    "id": "849b3151-25f7-41f3-b19b-fd8ad47bae18",
+    "overallStatus": "Not Delivered",
+    "timestamp": "2025-01-22T00:42:10.404Z",
+    "sender": "development.dev-elims",
+    "reportItemCount": 1,
+    "errorCount": 1,
+    "warningCount": 0,
+    "httpStatus": 201,
+    "destinations": [],
+    "reportId": "849b3151-25f7-41f3-b19b-fd8ad47bae18",
+    "topic": "full-elr",
+    "errors": [
+        {
+            "scope": "item",
+            "indices": [
+                1
+            ],
+            "trackingIds": [
+                "849b3151-25f7-41f3-b19b-fd8ad47bae18"
+            ],
+            "message": "Report could not processed. Reason: Duplicate report was detected and removed.",
```

Comment:
```suggestion
            "message": "Report could not processed. Reason: Duplicate report was detected and removed (no unique test results were found in the report.",
```

or some version of the above that sheds light on what a duplicate report actually is. This might be a good thing for @jsutantio to review.

Reply:
Ah thanks for the callout. I should have also pointed out the content of the warning/error messages when asked this morning what @jsutantio should review here. Personally, I don't love what I came up with, but I am somewhat ambiguous on what the right vocab is for the submission history API audience.